src/logic/resources-status.ts

```diff
@@ -0,0 +1,40 @@
+import { AppComponents } from '../types'
+import os from 'node-os-utils'
+
+export type IResourcesStatusComponent = {
+  areResourcesOverloaded: () => Promise<boolean>
+}
+
+function isAboveNinetyPercent(measure: number) {
+  return measure > 90
+}
+
+const getResourcersUsage = async (): Promise<{ cpuLoad: number; memLoad: number }> => {
+  const cpu = await os.cpu.usage()
+  const mem = await os.mem.used()
+
+  return { cpuLoad: cpu, memLoad: (mem.usedMemMb * 100) / mem.totalMemMb }
+}
+
+export function createResourcesStatusComponent(components: Pick<AppComponents, 'logs'>): IResourcesStatusComponent {
+  async function areResourcesOverloaded(): Promise<boolean> {
+    const logger = components.logs.getLogger('status-checker')
```

Comment:
I would move this to the outer function so you create it once for the whole component